bot/exts/info/code_snippets.py

```diff
@@ -0,0 +1,226 @@
+import re
+import textwrap
+from urllib.parse import quote_plus
+
+from discord import Message
+from discord.ext.commands import Cog
+
+from bot.bot import Bot
+from bot.utils.messages import wait_for_deletion
+
+
+GITHUB_RE = re.compile(
+    r'https://github\.com/(?P<repo>.+?)/blob/(?P<path>.+/.+)'
+    r'#L(?P<start_line>\d+)([-~]L(?P<end_line>\d+))?\b'
+)
+
+GITHUB_GIST_RE = re.compile(
+    r'https://gist\.github\.com/([^/]+)/(?P<gist_id>[^\W_]+)/*'
+    r'(?P<revision>[^\W_]*)/*#file-(?P<file_path>.+?)'
+    r'-L(?P<start_line>\d+)([-~]L(?P<end_line>\d+))?\b'
+)
+
+GITLAB_RE = re.compile(
+    r'https://gitlab\.com/(?P<repo>.+?)/\-/blob/(?P<path>.+/.+)'
+    r'#L(?P<start_line>\d+)([-](?P<end_line>\d+))?\b'
+)
+
+BITBUCKET_RE = re.compile(
+    r'https://bitbucket\.org/(?P<repo>.+?)/src/(?P<ref>.+?)/'
+    r'(?P<file_path>.+?)#lines-(?P<start_line>\d+)(:(?P<end_line>\d+))?\b'
+)
+
+
+class CodeSnippets(Cog):
+    """
+    Cog that parses and sends code snippets to Discord.
+
+    Matches each message against a regex and prints the contents of all matched snippets.
+    """
+
+    async def _fetch_response(self, url: str, response_format: str, **kwargs) -> str:
+        """Makes http requests using aiohttp."""
+        async with self.bot.http_session.get(url, **kwargs) as response:
+            if response_format == 'text':
+                return await response.text()
+            elif response_format == 'json':
+                return await response.json()
+
+    def _find_ref(self, path: str, refs: tuple) -> tuple:
+        """Loops through all branches and tags to find the required ref."""
+        # Base case: there is no slash in the branch name
+        ref = path.split('/')[0]
+        file_path = '/'.join(path.split('/')[1:])
```

Comment:
Make use of the maxsplit argument
```suggestion
        ref, file_path  = path.split('/', 1)
```